synapse/storage/databases/main/pusher.py

```diff
@@ -396,42 +396,57 @@ def _delete_pushers(txn) -> int:
     async def _remove_deleted_email_pushers(
         self, progress: dict, batch_size: int
     ) -> int:
-        """A background update that deletes all pushers for email addresses no longer
-        associated with a user.
+        """A background update that deletes all pushers for deleted email addresses.
+
+        In previous versions of synapse, when users deleted their email address, it didn't
+        also delete all the pushers for that email address. This background update removes
+        those to prevent unwanted emails
```

Comment:
You may wish to mention that this should only need to be run once when users upgrade to v1.42.0. Otherwise it may be easy to assume this background job is run every so often to clean up old pushers.